### PR TITLE
futures: no_std support

### PR DESCRIPTION
**Pull request by baloo**
## Motivation

Being able to trace futures in platforms that do not support stdlib would be nice.

### CODE REVIEW

File: tracing-futures/Cargo.toml [hunk 1]
```diff
@@ -18,17 +18,19 @@ keywords = ["logging", "profiling", "tracing", "futures", "async"]
 license = "MIT"
 
 [features]
-default = ["std-future"]
+default = ["std-future", "std"]
 futures-01 = ["futures_01"]
 futures-03 = ["std-future", "futures", "futures-task"]
 std-future = ["pin-project"]
+std = ["tracing/std"]
+unsafe_global = ["tracing/unsafe_global"]
```
Comment (hawkw) [inline tracing-futures/Cargo.toml]: I don't entirely understand why `tracing-futures` needs to expose its own feature flag for this?

Comment (baloo) [reply tracing-futures/Cargo.toml]: yeah, this should go in #497 and I should have splitted both more clearly.

Comment (baloo) [reply tracing-futures/Cargo.toml]: better this way.

**Comment by baloo** [comment]
that is unlikely to compile because it uses `dispatcher::with_default` from `tracing` which is not exposed until I find a solution to #497, ideally I'd wait, fix #497 first and then rebase this one atop of it.

**Comment by baloo** [comment]
tests wont pass because

**Comment by hawkw** [comment]
> that is unlikely to compile because it uses `dispatcher::with_default` from `tracing` which is not exposed until I find a solution to #497, ideally I'd wait, fix #497 first and then rebase this one atop of it.
> 
> error output:

If we're not able to find an acceptable solution for supporting `with_default` in a no-std context, we could always just make the `WithDispatcher`/`WithSubscriber` APIs require the `std` feature...if we did that, we could go ahead and merge this PR now, and remove the feature gating later (so that these APIs are enabled without `std`) later if #497 merges.

What do you think?

**Comment by baloo** [comment]
That would also work, but I don't mind waiting a couple more days see if we can get something working for #497

**Comment by hawkw** [comment]
@baloo actually, before we merge this, would you mind adding a line to the list of crate features in the docs? Thanks!

**Comment by baloo** [comment]
there

**Comment by hawkw** [comment]
merged, thanks for working on this!

**Pull request merged**